formats/Filtered/libs/ext.srf.filtered.value-filter.js

```diff
@@ -254,11 +262,14 @@
 				filtercontrols.height( height );
 			}
 
-
 			// insert options (checkboxes and labels) and attach event handlers
 			// TODO: Do we need to wrap these in a form?
-			for ( var j in sortedDistinctValues ) {
+			for (var j = 0; j < sortedDistinctValues.length; j++) {
 				var option = $('<div class="filtered-value-option">');
+
+				// Try to get label, if not fall back to value id
+				var label = distinctValues[sortedDistinctValues[j]] || sortedDistinctValues[j];
+
 				var checkbox = $('<input type="checkbox" class="filtered-value-value" value="' + sortedDistinctValues[j] + '"  >');
```

Comment:
This function is huge. Very procedural programming style. I'm ranting about code quality here. (Which you can disregard if you want, I'm not objecting to this being merged.)